homeassistant/components/ambient_station/sensor.py

```diff
@@ -337,162 +357,189 @@
         name="Soil Temp 10",
         native_unit_of_measurement=TEMP_FAHRENHEIT,
         device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
     ),
     SensorEntityDescription(
         key=TYPE_SOILTEMP1F,
         name="Soil Temp 1",
         native_unit_of_measurement=TEMP_FAHRENHEIT,
         device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
     ),
     SensorEntityDescription(
         key=TYPE_SOILTEMP2F,
         name="Soil Temp 2",
         native_unit_of_measurement=TEMP_FAHRENHEIT,
         device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
     ),
     SensorEntityDescription(
         key=TYPE_SOILTEMP3F,
         name="Soil Temp 3",
         native_unit_of_measurement=TEMP_FAHRENHEIT,
         device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
     ),
     SensorEntityDescription(
         key=TYPE_SOILTEMP4F,
         name="Soil Temp 4",
         native_unit_of_measurement=TEMP_FAHRENHEIT,
         device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
     ),
     SensorEntityDescription(
         key=TYPE_SOILTEMP5F,
         name="Soil Temp 5",
         native_unit_of_measurement=TEMP_FAHRENHEIT,
         device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
     ),
     SensorEntityDescription(
         key=TYPE_SOILTEMP6F,
         name="Soil Temp 6",
         native_unit_of_measurement=TEMP_FAHRENHEIT,
         device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
     ),
     SensorEntityDescription(
         key=TYPE_SOILTEMP7F,
         name="Soil Temp 7",
         native_unit_of_measurement=TEMP_FAHRENHEIT,
         device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
     ),
     SensorEntityDescription(
         key=TYPE_SOILTEMP8F,
         name="Soil Temp 8",
         native_unit_of_measurement=TEMP_FAHRENHEIT,
         device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
     ),
     SensorEntityDescription(
         key=TYPE_SOILTEMP9F,
         name="Soil Temp 9",
         native_unit_of_measurement=TEMP_FAHRENHEIT,
         device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
     ),
     SensorEntityDescription(
         key=TYPE_SOLARRADIATION,
         name="Solar Rad",
         native_unit_of_measurement=IRRADIATION_WATTS_PER_SQUARE_METER,
         device_class=DEVICE_CLASS_ILLUMINANCE,
+        state_class=STATE_CLASS_MEASUREMENT,
     ),
     SensorEntityDescription(
         key=TYPE_SOLARRADIATION_LX,
         name="Solar Rad (lx)",
```

Comment:
Drop the unit from the name.

Reply:
But then won't the entity ID for this be something like `sensor.solar_rad_2`? That doesn't really describe what this is intended to calculate and show.￼￼

Reply:
then set the entity ID on the instance before passing it to `add_entities` and it will be used as a suggestion. Don't mess with the name to steer entity ID.

Reply:
@balloob Like this? https://github.com/home-assistant/core/pull/55412/commits/1b9e3cd55da0c14e1d98edca96e6c8e136c46b45

Reply:
yes

Reply:
That will do :)